modules/nextflow/src/main/groovy/nextflow/Nextflow.groovy

```diff
@@ -135,7 +135,11 @@ class Nextflow {
         }
 
         // revolve the glob pattern returning all matches
-        return fileNamePattern(splitter, options)
+        final result = fileNamePattern(splitter, options)
+        if( glob && options?.checkIfExists && result.isEmpty() )
+            throw new NoSuchFileException(path.toString())
```

Comment:
Maybe, we could add a new case in the [NexflowTest.'should check if file exists'](https://github.com/nextflow-io/nextflow/blob/aacb69f8b6ff55d1cb09cb53c6ac5e41781ab6a6/modules/nextflow/src/test/groovy/nextflow/NextflowTest.groovy#L350) to cover this case. Something like this
```
when:
        Nextflow.file("$folder/*.txt", checkIfExists: true)
 then:
        e = thrown(NoSuchFileException)
        e.message == "$folder/*.txt".toString()
```

Reply:
Good catch